packages/gatsby/src/state-machines/actions.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  assign,
+  AnyEventObject,
+  ActionFunction,
+  spawn,
+  ActionFunctionMap,
+  DoneEventObject,
+} from "xstate"
+import { Store } from "redux"
+import { IBuildContext, IMutationAction } from "../services"
+import { actions } from "../redux/actions"
+import { listenForMutations } from "../services/listen-for-mutations"
+import { DataLayerResult } from "./data-layer"
+import { assertStore } from "../utils/assert-store"
+
+export const callRealApi = async (
+  event: IMutationAction,
+  store?: Store
+): Promise<unknown> => {
```

Comment:
This function is not explicitly returning a Promise so it returns an implicit Promise that resolves to the return value `null`

```suggestion
): Promise<null> => {
```

Reply:
Yes, this doesn't need to be async at all in fact.